### PR TITLE
sql: add unimplemented support for alter type commands

### PR DESCRIPTION
**Pull request by rohany**
This PR adds unimplemented support and necessary boilerplate
for some of the ALTER TYPE commands.

Release note: None

### CODE REVIEW

File: pkg/sql/alter_type.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
+	"github.com/cockroachdb/errors"
+)
+
+type alterTypeNode struct {
+	n *tree.AlterType
+}
+
+// Use to satisfy the linter.
```
Comment (otan) [inline pkg/sql/alter_type.go:25]: nit: o_o maybe a nicer comment is "alterTypeNode implements planNode"

File: pkg/sql/parser/sql.y [hunk 2]
```diff
@@ -1867,6 +1873,65 @@ opt_validate_behavior:
     $$.val = tree.ValidationDefault
   }
 
+// %Help: ALTER TYPE - change the definition of a type.
+// %Category: DDL
+// %Text: ALTER TYPE <typename> <command>
+//
+// Commands:
+//   ALTER TYPE ... ADD VALUE <value> [ { BEFORE | AFTER } <value> ]
```
Comment (otan) [inline pkg/sql/parser/sql.y:1881]: i'm looking at https://www.postgresql.org/docs/current/sql-altertype.html and we seem to be missing a few. should we add those in?

Comment (rohany) [reply pkg/sql/parser/sql.y:1881]: I left them out because I don't think we have any upcoming plans to implement them (I can if you think I should though)

Comment (otan) [reply pkg/sql/parser/sql.y:1881]: worth adding for metricss' sake, at the very least..

File: pkg/sql/alter_type.go (hunk 1 above)
Comment (rohany) [reply pkg/sql/alter_type.go]: done

File: pkg/sql/parser/sql.y (hunk 2 above)
Comment (rohany) [reply pkg/sql/parser/sql.y]: done

wew that was a bunch of extra yaccing around

File: pkg/sql/parser/sql.y [hunk 3]
```diff
@@ -600,7 +603,7 @@ func (u *sqlSymUnion) typeReferences() []tree.ResolvableTypeReference {
 %token <str> NONE NORMAL NOT NOTHING NOTNULL NOWAIT NULL NULLIF NULLS NUMERIC
 
 %token <str> OF OFF OFFSET OID OIDS OIDVECTOR ON ONLY OPT OPTION OPTIONS OR
-%token <str> ORDER ORDINALITY OTHERS OUT OUTER OVER OVERLAPS OVERLAY OWNED OPERATOR
+%token <str> ORDER ORDINALITY OTHERS OUT OUTER OVER OVERLAPS OVERLAY OWNED OWNER OPERATOR
```
Comment (otan) [inline pkg/sql/parser/sql.y:606]: haha this is definitely already out of order.

**Review by otan — APPROVED** [review]

File: pkg/sql/parser/sql.y (hunk 3 above)
Comment (rohany) [reply pkg/sql/parser/sql.y:606]: F

**Pull request merged**